### ISSUE TITLE
Image size problem

### ISSUE THREAD

**Issue by JanceeLee** [torrvision/crfasrnn#60]
Hi,
Thanks for your great work on DL.I use the latest FCN code plus the CRF layer, and get the following error at run time：

I0722 23:26:08.296314 27585 net.cpp:274] Network initialization done.
I0722 23:26:08.296448 27585 solver.cpp:60] Solver scaffolding done.
libprotobuf WARNING google/protobuf/io/coded_stream.cc:478] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
I0722 23:26:50.454507 27585 net.cpp:752] Ignoring source layer loss
I0722 23:26:50.658272 27585 eltwise_layer.cpp:39] 1  4 16 14 ；1  4 16 14
I0722 23:26:50.659008 27585 eltwise_layer.cpp:39] 1  4 34 30 ；1  4 34 30
I0722 23:26:50.681608 27585 eltwise_layer.cpp:39] 1  4 209 180 ；1  4 209 180
I0722 23:26:50.681612 27585 eltwise_layer.cpp:39] 1  4 209 180 ；1  4 209 180
I0722 23:26:50.681617 27585 eltwise_layer.cpp:39] 1  4 209 180 ；1  4 209 180
I0722 23:26:50.694358 27585 eltwise_layer.cpp:33] 1  4 209 180 ；1  4 209 180
I0722 23:26:50.694373 27585 eltwise_layer.cpp:39] 1  4 209 180 ；1  4 209 180
I0722 23:26:50.809535 27585 solver.cpp:228] Iteration 0, loss = 1994.94
I0722 23:26:50.809564 27585 solver.cpp:244]     Train net output #0: loss = 1994.94 (\* 1 = 1994.94 loss)
I0722 23:26:50.809572 27585 sgd_solver.cpp:106] Iteration 0, lr = 1e-10
I0722 23:26:51.226410 27585 eltwise_layer.cpp:39]1  4 22 28 ；1  4 22 28
I0722 23:26:51.228551 27585 eltwise_layer.cpp:39] 1  4 46 58 ；1  4 46 58
I0722 23:26:51.265045 27585 eltwise_layer.cpp:39] 1  4 209 180 (bottom[1]->shape()) ； 1 4 314 407 (bottom[0]->shape())
**F0722 23:26:51.265060 27585 eltwise_layer.cpp:40] ## Check failed: bottom[i]->shape() == bottom[0]->shape()**

I hope you can answer the following questions：
Is it because I didn't resize the image to 500＊500？
How to train arbitary image,just like FCN?

Thanks

**Comment by 24hours** [comment]
Can provide the train.prototxt you are using  ?
the error come from https://github.com/bittnt/caffe/blob/master/src/caffe/layers/meanfield_iteration.cpp#L166 . 

This error only occur when caffe is processing image 2 and Image 1 , 2 having different dimension. When meanfield are being iterated, it somehow capture image 1 dimension instead of using  image 2 dimension.

I'm trying to tackle this problem in my implementation as well. 
However, I have modified my train.txt to use python data layer, I'm not sure the error persist if LMDB is used. I'm too lazy to setup LMDB.

**Comment by 24hours** [comment]
it seems like meanfield layer required fixed size image input, pad / resize data is the solution here.

**Comment by sadeepj** [comment]
Closing old issues with no recent activity.
